redis/src/main/scala/zio/redis/executor/RedisConnection.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis.executor
+
+import zio._
+import zio.redis.{RedisError, RedisUri, logScopeFinalizer}
+import zio.stream.{Stream, ZStream}
+
+import java.io.{EOFException, IOException}
+import java.net.{InetSocketAddress, SocketAddress, StandardSocketOptions}
+import java.nio.ByteBuffer
+import java.nio.channels.{AsynchronousSocketChannel, Channel, CompletionHandler}
+
+private[redis] trait RedisConnection {
+  def read: Stream[IOException, Byte]
+  def write(chunk: Chunk[Byte]): IO[IOException, Option[Unit]]
+}
+
+private[redis] final class RedisConnectionLive(
```

Comment:
Let's keep the structure that was present before, i.e. one top-level artifact per file.